extensions/cl_arm_printf.asciidoc

```diff
@@ -0,0 +1,192 @@
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_arm_printf
+:R: pass:q,r[^(R)^]
+
+== Name Strings
+
+`cl_arm_printf`
+
+== Contact
+
+Kevin Petit, Arm (kevin.petit 'at' arm.com)
+
+== Contributors
+
+Scott Moyers, Arm Ltd. +
+Robert Elliott, Arm Ltd. +
+Mats Petersson, Arm Ltd. +
+Vatsalya Prasad, Arm Ltd. +
+Kevin Petit, Arm Ltd. +
+
+== Notice
+
+Copyright (c) 2014-2021 Arm Ltd.
+
+== Status
+
+Shipping.
+
+== Version
+
+Built On: {docdate} +
+Version: 2.0.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification Version 3.0.10.
+
+This extension requires OpenCL 1.0 (version 1.0.0) or OpenCL 1.2 (version 2.0.0).
+
+== Overview
+
+This extension enables the device side `printf` built in function for OpenCL C
+versions prior to 1.2 (version 1.0.0 only).
+
+It also extends the `cl_context_properties` enumeration to allow a user defined
+printf callback and/or printf buffer size.
+
+The `printf` built-in function should be used for debugging purposes only and may
+have a significant negative impact on the performance of an OpenCL application.
+
+== New API Enums
+
+Accepted value for the _properties_ parameter to *clCreateContext*:
+
+[source,c]
+----
+CL_PRINTF_CALLBACK_ARM    0x40B0
+CL_PRINTF_BUFFERSIZE_ARM  0x40B1
+----
+
+== New OpenCL C Functions
+
+In version 1.0.0 only:
+
+[source,c]
+----
+int printf( constant char * restrict format, ... );
```

Comment:
I didn't see this PR until now, so my apologies for commenting on this after the PR has been merged: I notice that the `const` qualifier is missing here.